docs/en/reference/inheritance-mapping.rst

```diff
@@ -153,6 +153,20 @@ lowercase short name of each class as key.
     to see if they are part of it. The resulting map, however, can be kept
     in the metadata cache.
 
+Performance impact on to-one associations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There is a general performance consideration when using entity inheritance:
+If the target-entity of a many-to-one or one-to-one association is part of
+an inheritance hierarchy, it is preferable for performance reasons that it
+be a leaf entity (ie. have no subclasses).
+
+Otherwise, the ORM is currently unable to tell beforehand which entity class
+will have to be used, and so no appropriate proxy instance can be created.
+That means the referred-to entities will *always* be loaded eagerly, which
+might even include other entities as well (in the case of self-referencing
+associations).
```

Comment:
> which might even include other entities as well (in the case of self-referencing
associations).



I don't understand what that last bit is about… "other entities" than what?

Reply:
When the referred-to entity that is loaded eagerly has other associations of the same kind, it may trigger another round of eager loads 🔁

Reply:
By "of the same kind", you mean "self-referencing", right?

If yes, I'd go with "That means the referred-to entities will always be loaded eagerly, which might even propagate to relationships of these entities (in the case of self-referencing associations).

Reply:
:+1:, changed that.